tiny_dnn/core/framework/tensor.h

```diff
@@ -348,6 +353,22 @@ class Tensor {
         std::copy(shape.begin(), shape.end(), shape_.begin());
     }
 
+    Tensor view_impl(std::initializer_list<size_t> const &start,
+                     std::initializer_list<size_t> const &new_shape) {
+	if (start.size() > kDimensions || new_shape.size() > kDimensions) {
+            throw nn_error("Overpassed number of existing dimensions.");
+	}
+
+	// compute the new offset and check that it's feasible to create
+	// the new view.
+	const size_t new_offset = compute_offset(start, shape_);
+	if (new_offset + product(new_shape) > size_) {
+            throw nn_error("Cannot create a view from this tensor");
+	}
+
+        return Tensor(storage_ptr_, new_offset, new_shape);
```

Comment:
Let's make a simple example with a 3x3 tensor, and a 2x2 subtensor.

The storage is:
1 2 3 4 5 6 7 8 9

And the value for the offset and shape are
offset = 0, shape = {3, 3}

Now, I want to take a subtensor, in different ways:

1) first row
2) second row,
3) first column
4) top-left 2x2 submatrix

1) Offset = 0, shape = {1, 3} <- ok: {{1, 2, 3}}
2) Offset = 3, shape = {1, 3} <- ok: {{1, 2, 3}}
3) Offset = 0, shape = {3, 1} <- Broken! It will result {{1}, {2}, {3}}
4) Offset = 0, shape = {2, 2} <- Broken! It will result {{1, 2}, {3, 4}}

Am I missing something obvious?